packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -396,6 +419,20 @@ public void isSignedIn(final Result result) {
       result.success(value);
     }
 
+    @Override
+    public void recoverAuth(Result result, String accountId) {
+      if (userRecoverableAuthIntent == null) {
+        throw new IllegalStateException("No recoverable auth intent available.");
+      } else if (currentAccount != null && !accountId.equals(currentAccount.getId())) {
+        throw new IllegalStateException("No recoverable auth intent for this account.");
```

Comment:
Instead of throwing, can we make the result a failure? I would prefer to not cause the app to crash.

Reply:
I changed it to throw a PlatformException instead. This check is to just make sure the developer doesn't sign in -> get UserRecoverAuthException -> log out -> try to start intent and recover Auth. 

Also, `recoverAuth()` now returns a boolean to indicate if the User took successful actions to recover auth.